File: leetcode/e3340_check_balanced_string/Solution.java
package e3340_check_balanced_string;

class Solution {
    public boolean isBalanced(String num) {
        int odd = 0;
        int even = 0;
        for (int i = 0; i < num.length(); i++) {
            if (i % 2 == 1) {
                odd = odd + (num.charAt(i) - '0');
            } else {
                even = even + (num.charAt(i) - '0');
            }
        }
        return odd == even;
    }
}

File: leetcode/e3340_check_balanced_string/solution.go
package main

func isBalanced(s string) bool {
	odd := 0
	even := 0
	for i := 0; i < len(s); i++ {
		if i%2 == 0 {
			even += int(s[i] - '0')
		} else {
			odd += int(s[i] - '0')
		}
	}
	return odd == even
}
